packages/server/src/utils/databases/mongo.ts

```diff
@@ -28,48 +28,56 @@ export const buildMongo = async (mongo: MongoNested) => {
 		databasePassword,
 		command,
 		mounts,
+		replicaSets,
 	} = mongo;
 
-	const initReplicaSet = `
+	const startupScript = `
 #!/bin/bash
+${
+	replicaSets
+		? `
 mongod --port 27017 --replSet rs0 --bind_ip_all &
 MONGOD_PID=$!
 
 # Wait for MongoDB to be ready
 while ! mongosh --eval "db.adminCommand('ping')" > /dev/null 2>&1; do
-  sleep 2
+	sleep 2
 done
 
 # Check if replica set is already initialized
 REPLICA_STATUS=$(mongosh --quiet --eval "rs.status().ok || 0")
 
 if [ "$REPLICA_STATUS" != "1" ]; then
-  echo "Initializing replica set..."
-  mongosh --eval '
-    rs.initiate({
-      _id: "rs0",
-      members: [{ _id: 0, host: "localhost:27017", priority: 1 }]
-    });
+	echo "Initializing replica set..."
+	mongosh --eval '
+	rs.initiate({
+		_id: "rs0",
+		members: [{ _id: 0, host: "localhost:27017", priority: 1 }]
+	});
 
     // Wait for the replica set to initialize
-    while (!rs.isMaster().ismaster) {
-      sleep(1000);
-    }
+	while (!rs.isMaster().ismaster) {
+		sleep(1000);
+	}
 
     // Create root user after replica set is initialized and we are primary
-    db.getSiblingDB("admin").createUser({
-      user: "${databaseUser}",
-      pwd: "${databasePassword}",
-      roles: ["root"]
-    });
-  '
+	db.getSiblingDB("admin").createUser({
+		user: "${databaseUser}",
+		pwd: "${databasePassword}",
+		roles: ["root"]
+	});
+	'
+
 else
-  echo "Replica set already initialized."
+	echo "Replica set already initialized."
 fi
+`
+		: "mongod --port 27017 --bind_ip_all & MONGOD_PID=$!"
+}
 
-wait $MONGOD_PID`;
+${command ?? "wait $MONGOD_PID"}`;
 
-	const defaultMongoEnv = `MONGO_INITDB_DATABASE=admin\n${env ? `${env}` : ""}`;
+	const defaultMongoEnv = `MONGO_INITDB_ROOT_USERNAME=${databaseUser}\nMONGO_INITDB_ROOT_PASSWORD=${databasePassword}\nMONGO_INITDB_DATABASE=admin\n${env ? `${env}` : ""}`;
```

Comment:
I left the username and password env variables here at all times, but they will only be actually used when the non-replica set mode is used because the replica set initialization script does it otherwise.